Desktop troubleshoot topic (#16427)
docker/docs: desktop/troubleshoot/topics.md

Volume mounting requires shared drives for projects that live outside of the
`/home/<user>` directory. From **Settings**, select **Resources** and then **File sharing**. Share the drive that contains the Dockerfile and volume.

### Docker Desktop fails to start on MacOS or Linux platforms

> On MacOS and Linux, Docker Desktop creates unix domain sockets for communication among various docker processes.

Reviewer: ```suggestion
On MacOS and Linux, Docker Desktop creates Unix domain sockets used for inter-process communication.
```

> Docker fails to start if the absolute path length of any of these sockets exceeds the OS limitation which is 104 characters on MacOS and 108 characters on Linux. These sockets are created under the user's home directory, if the user id length is such that the absolute path of the socket exceeds the OS path length limitation, then Docker Desktop is unable to create the socket and fails to start. The workaround for this is to shorten the user id.

Reviewer: ```suggestion
Docker fails to start if the absolute path length of any of these sockets exceeds the OS limitation which is 104 characters on MacOS and 108 characters on Linux. These sockets are created under the user's home directory. If the user ID length is such that the absolute path of the socket exceeds the OS path length limitation, then Docker Desktop is unable to create the socket and fails to start. The workaround for this is to shorten the user ID.
```

Reviewer: Also - shouldn't we be able to say the length of the longest user ID that we allow?

> Docker fails to start if the absolute path length of any of these sockets exceeds the OS limitation which is 104 characters on MacOS and 108 characters on Linux. These sockets are created under the user's home directory. If the user ID length is such that the absolute path of the socket exceeds the OS path length limitation, then Docker Desktop is unable to create the socket and fails to start. The workaround for this is to shorten the user ID whic we recommend has a maximum length of 33 characters on MacOS and 55 characters on Linux. No need to specify the minimum length as that is governed by the OS.

Reviewer: ```suggestion
Docker fails to start if the absolute path length of any of these sockets exceeds the OS limitation which is 104 characters on MacOS and 108 characters on Linux. These sockets are created under the user's home directory. If the user ID length is such that the absolute path of the socket exceeds the OS path length limitation, then Docker Desktop is unable to create the socket and fails to start. The workaround for this is to shorten the user ID which we recommend has a maximum length of 33 characters on MacOS and 55 characters on Linux. No need to specify the minimum length as that is governed by the OS.
```
